### PR TITLE
fix issue 16192 - std.conv.toChars() opSlice wrong for radix other th…

### PR DESCRIPTION
**Pull request by aG0aep6G**
…an 10

https://issues.dlang.org/show_bug.cgi?id=16192

Was #4457. This one targets master and adds more extensive tests of `opSlice`.

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -5692,7 +5692,7 @@ auto toChars(ubyte radix = 10, Char = char, LetterCase letterCase = LetterCase.l
             enum SHIFT = 4;
         else
             static assert(0);
-        struct Result
+        static struct Result
```
Comment (joakim-noah) [inline std/conv.d]: Why is this needed?

Comment (aG0aep6G) [inline std/conv.d]: Without "static" the struct gets an unneeded context pointer, because it's declared in a function. opSlice does void initialization and doesn't copy the context pointer over, so comparisons would fail.

Since the context pointer isn't needed, getting rid of it seems to be obvious solution here. The base 10 Result above is also static.

Comment (joakim-noah) [inline std/conv.d]: OK, I didn't know what the `static` keyword meant in this context or why your tests needed it, got it now.

**Comment by aG0aep6G** [comment]
This has been idle for a week. Can I get some review on this?

**Comment by quickfur** [comment]
LGTM.

**Comment by quickfur** [comment]
Auto-merge toggled on
